src/clarity-angular/popover/signpost/signpost.ts

```diff
@@ -152,10 +145,27 @@ export class Signpost {
      */
     public signpostOptions: PopoverOptions = {
         allowMultipleOpen: false,
-        offsetY: this._offsetY, //TODO: set these based on signPostPosition case
-        offsetX: this._offsetX
+        offsetY: 0, //TODO: set these based on signPostPosition case
+        offsetX: 0
     };
 
+
+    constructor( public ifOpenService: IfOpenService ) {
+        this.openSubscription = ifOpenService.openedChange.subscribe(change => {
+            this.open = change;
+        });
+    }
+
+    /*********
+     * @function ngAfterContentInit
+     *
+     * @description
+     * Angular lifecycle hook to set the signpost trigger
+     */
+    ngAfterContentInit() {
+        this.setTrigger(this.signpostTrigger);
+    }
```

Comment:
So this whole `setTrigger()` in `ngAfterContentInit()` thing is a bit obfuscated, and as such it has a bug: if I start by providing a custom trigger, then remove it (for instance with `*ngIf`), the `useCustomTrigger` property will never revert back to false.

There is a much more straightforward solution:
```typescript
public useCustomTrigger: boolean = false;

@ContentChild(SignpostTriggerDirective) 
set customTrigger(trigger: SignpostTriggerDirective) {
  this.useCustomTrigger = !!trigger;
}
```